File: PudimDouroAPI/src/main/java/br/com/mountainfortress/PudimDouroAPI/model/Edition.java
package br.com.mountainfortress.PudimDouroAPI.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Year;

@Entity
@Table(name = "edition")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Edition {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    private Year edition_year;

    @NotNull
    private boolean active;

    private Integer first_place;

    private Integer second_place;

    private Integer third_place;
}

File: pudim_douro/lib/models/edition.dart
class Edition {
  final String id;
  final String editionYear;
  final String active;
  final String firstPlace;
  final String secondPlace;
  final String thirdPlace;
  bool isExpanded;

  Edition(
    this.id,
    this.editionYear,
    this.active,
    this.firstPlace,
    this.secondPlace,
    this.thirdPlace,
    this.isExpanded,
  );

  Edition.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        editionYear = json['edition_year'],
        active = json['active'],
        firstPlace = json['first_place'],
        secondPlace = json['second_place'],
        thirdPlace = json['third_place'],
        isExpanded = false;

  void setIsExpanded(bool isExpanded){
    this.isExpanded = isExpanded;
  }
}